src/Streamly/Internal/Data/Fold.hs

```diff
@@ -450,6 +434,20 @@ transform (Pipe pstep1 pstep2 pinitial) (Fold fstep finitial fextract) =
 
     extract (Tuple' _ fs) = fextract fs
 
+-- | @mapMaybe f fold@ maps a 'Maybe' returning function @f@ on the input of
+-- the fold, filters out 'Nothing' elements, and return the values extracted
+-- from 'Just'.
+--
+-- >>> f x = if even x then Just x else Nothing
+-- >>> fld = Fold.mapMaybe f Fold.toList
+-- >>> Stream.fold fld (Stream.enumerateFromTo 1 10)
+-- [2,4,6,8]
+--
+-- /Internal/
+{-# INLINE mapMaybe #-}
+mapMaybe :: (Monad m) => (a -> Maybe b) -> Fold m b r -> Fold m a r
+mapMaybe f = map f . filter isJust . map fromJust
+
```

Comment:
Why was this moved here?

Reply:
This was in the wrong place, to begin with. This was grouped with `rmapM` and `rsequence`.